File: fixtures/scale/python/large/cipherlab/checks/check_sparkstream.py
from cipherlab.ciphers.sparkstream import sparkstream_encrypt, sparkstream_decrypt
from cipherlab.specs.sparkstream_spec import sparkstream_spec
from cipherlab.corpus.corpus_stream import corpus_stream


def check_sparkstream():
    spec = sparkstream_spec()
    for text in corpus_stream():
        sealed = sparkstream_encrypt(text, spec["key"])
        opened = sparkstream_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "stream"

File: fixtures/scale/ruby/large/lib/check_sparkstream.rb
require_relative 'sparkstream_encrypt'
require_relative 'sparkstream_decrypt'
require_relative 'corpus_stream'

def check_sparkstream
  corpus_stream.each do |text|
    sealed = sparkstream_encrypt(text, 15)
    opened = sparkstream_decrypt(sealed, 15)
    return false if opened != text
  end
  true
end
